server/src/main/java/org/opensearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -219,14 +221,27 @@ protected TransportReplicationAction(
 
         transportService.registerRequestHandler(actionName, ThreadPool.Names.SAME, requestReader, this::handleOperationRequest);
 
-        transportService.registerRequestHandler(
-            transportPrimaryAction,
-            executor,
-            forceExecutionOnPrimary,
-            true,
-            in -> new ConcreteShardRequest<>(requestReader, in),
-            this::handlePrimaryRequest
-        );
+        // Register only TransportShardBulkAction for admission control ( primary indexing action )
+        if (transportPrimaryAction.equals(TransportShardBulkAction.ACTION_NAME + PRIMARY_ACTION_SUFFIX)) {
```

Comment:
The problem I see with this logic is error prone because
1. The else part will register a non shard bulk or non primary action once and re-register again in the next part of the ctor execution
2. Couldn't we use the `forceExecutionOnPrimary ` flag to apply admission control instead of re-doing the logic here? This already takes care of all other actions as well without the need to explicitly handle them.

Reply:
Can we use "forceExecution" and remove if else. Different callers of this method has forceExecution as true which skips rejection.

So even if they're registered with "admissionControlActionType", if "forceExection" is true, then we will skip admission control otherwise we'll apply admission control.

Reply:
> Couldn't we use the canTripCircuitBreaker  flag to apply admission control instead of re-doing the logic here? This already takes care of all other actions as well without the need to explicitly handle them.

`canTripCircuitBreaker` is true for all actions in `TransportReplicationAction.java`  but there is forceExecutionFlag which will skip circuit breaker. So we can remove the if else check if we rely on forceExecutionFlag.
if forceExecution is false we will apply admissionControl on transport actions and it is same behaviour for circuit breaker as well. Let me know if this works? 

If we implemented this way along with indexingPrimaryAction we will apply admissionControl to other actions for which circuit breaker is applied and AC will be inline with circuit breaker.